server/src/main/java/com/thoughtworks/go/config/update/ElasticAgentProfileUpdateCommand.java

```diff
@@ -52,6 +52,11 @@ public boolean canContinue(CruiseConfig cruiseConfig) {
         return super.canContinue(cruiseConfig) && isRequestFresh(cruiseConfig);
     }
 
+    @Override
+    public void encrypt(CruiseConfig preProcessedConfig) {
+        this.elasticProfile.encryptSecureProperties(preProcessedConfig);
```

Comment:
add some tests to verify encryption at the EntityConfigUpdateCommand (during update)